src/components/features/carousel/Carousel.jsx

```diff
@@ -0,0 +1,98 @@
+import { useMemo, useState } from "react";
+import CarouselSlide from "./CarouselSlide";
+import CarouselButtonLeft from "./CarouselButtonLeft";
+import CarouselButtonRight from "./CarouselButtonRight";
+
+/**
+ * Carousel Component 입니다. height속성을 갖는 wrapper를 필요로 합니다.
+ * @param slides carousel에 들어갈 slides, slides의 형태는 다음과 같습니다.
+ * [{image: "image url",to: "link to"},...]
+ * @returns {JSX.Element}
+ * @constructor
+ */
+const Carousel = ({ slides }) => {
+  const [currentSlide, setCurrentSlide] = useState(1);
+
+  // 들어온 slides를 토대로 edge slide를 추가합니다.
+  const mountedSlides = useMemo(() => {
+    if (slides.length === 1) {
+      return [...slides, ...slides, ...slides];
+    } else if (slides.length === 2) {
+      return [...slides, ...slides];
+    } else {
+      return [slides[slides.length - 1], ...slides, slides[0]];
+    }
+  }, [slides]);
+
+  const length = useMemo(() => mountedSlides.length, [mountedSlides.length]);
+
+  const prevSlide = useMemo(
```

Comment:
useMemo 로 굳이 메모아이징 하지 않아도 되겠네요
```suggestion
  const prevSlide = 
```